omniqueue/src/lib.rs

```diff
@@ -1,3 +1,101 @@
+//! # Omniqueue
+//!
+//! Omniqueue is an abstraction layer over various queue backends. As of present it supports:
+//!
+//!   * In-memory queues
+//!
+//!   * RabbitMQ
+//!
+//!   * Redis streams
+//!
+//!   * SQS
+//!
+//! Omniqueue provides a high-level interface for sending and receiving
+//!
+//!   * Raw byte arrays in the way most compatible with the queue backend
+//!
+//!   * JSON encoded byte arrays for types that implement [`serde::Deserialize`] and
+//!     [`serde::Serialize`]
+//!
+//!   * Arbitrary types for which an encoder and/or decoder has been defined
+//!
+//! ## How to Use Omniqueue
+//!
+//!   1. Start by adding `omniqueue` to your `Cargo.toml`. If you don't need support for all of the
+//!      backends, disable the default features and enable each required backend individually.
```

Comment:
I might hoist this out of the "how to use" and make it a dedicated section titled `Cargo Features` or similar. It could take the place of the list of supported backends up top (since it expresses the same information).

Add a remark about "all enabled by default" and that's probably all you need.

Reply:
To emphasize, I don't think it's important to say "add this to your `Cargo.toml`" here. Looking at other projects, they sometimes just show a tiny part of the manifest at the head of the readme.

Looking at [serde_json](https://github.com/serde-rs/json) as an example, they show:

```toml
[dependencies]
serde_json = "1.0"
```

with zero explanation. Heading, badges, short description, then this fragment of the manifest.

Either way, it probably isn't needed in the rustdoc. If someone is reading the rustdoc, they probably already have an idea of how to install the crate.